torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2290,12 +2313,18 @@ def forward(self, *args: Any, **kwargs: Any) -> Any:
                     input_dtype, *args, **kwargs
                 )
 
-            # All-gather full parameters, moving them to compute_device if
-            # necessary.
-            self._rebuild_full_params()
+            # Only rebuilding full params when the params are not prefetched in previous layers
+            if not self._forward_full_params_prefetched:
```

Comment:
'_forward_full_params_prefetched' is only set and visible for current layer, it can not know previous layer's flag and check it here

Reply:
Synced up offline, _forward_full_params_prefetched of the next layer is set at the current layer.